src/formats/formats.js

```diff
@@ -37,40 +37,42 @@ export function formatByExtension(extension) {
   }[extension];
 }
 
-function formatByName(name) {
+function formatByName(name, customDelimiter) {
   return {
     yml: yamlFormatter,
     yaml: yamlFormatter,
     toml: tomlFormatter,
     json: jsonFormatter,
     frontmatter: FrontmatterInfer,
-    'json-frontmatter': FrontmatterJSON,
-    'toml-frontmatter': FrontmatterTOML,
-    'yaml-frontmatter': FrontmatterYAML,
+    'json-frontmatter': frontmatterJSON(customDelimiter),
+    'toml-frontmatter': frontmatterTOML(customDelimiter),
+    'yaml-frontmatter': frontmatterYAML(customDelimiter),
```

Comment:
I do wonder if there is a better way to avoid the overhead that is incurred here. The benefit of the single objects that we had before was that there was only a single instance of each created for the entire CMS (7 total instances). Now, we're creating three new objects for each *page*, then, throwing them all away (unless one is used). That just seems a little wasteful to me.

To pass a custom parameter, we're obviously going to have to do more than one instance. I think the cleanest, most sustainable way would be to create only one instance per *collection*.
For now, I wonder if we could just wait to pass the `customDelimiter` in until after `formatByName` returns the function. I haven't thought this through that much, though.

@erquhart Do you think this could cause performance problems now, or would you like to merge this, and change it later? I haven't worked with JS enough to know how expensive this really is.

Reply:
I doubt any real performance penalty would happen here. I did have the same thoughts on improvements, but trying to prioritize timely merges over ideal code. This passed the "good enough" test for me, personally.

Reply:
So because I'm passing in customDelimiters here, you're saying that each object gets actually instantiated while before it did not? That totally escaped me, I can test out a few things and make this more optimized if you'd like.

Reply:
@Swieckowski I think, like Shawn said, that it's probably not worth delaying the PR just to try to optimize the code. The difference is that a new object gets created with each function call, instead of just reusing the same one. I don't really want to delay it just for that, and we can optimize later if necessary. Do you agree?

Reply:
Sure! I was just trying to make sure I was following your logic correctly.